tests/integration/eventengine/juniper_alert_count_test.py

```diff
@@ -0,0 +1,430 @@
+            in caplog.text
+        )
+        with pytest.raises(Event.DoesNotExist):
+            end_event.refresh_from_db()
+        assert getattr(handler, "event", None)
+
+
+########################
+#                      #
+# fixtures and helpers #
+#                      #
+########################
+def post_fake_yellow_alarm_start_event(netbox, count: int = 2):
+    event = Event(
+        source_id="ipdevpoll",
+        target_id="eventEngine",
+        event_type_id="juniperYellowAlarmState",
+        netbox=netbox,
+        state=Event.STATE_START,
+    )
+    event.varmap = {
+        "count": count,
+        "alerttype": "juniperYellowAlarmOn",
+    }
+    event.save()
+
+    return event
+
+
+def post_fake_yellow_alarm_end_event(netbox):
+    event = Event(
+        source_id="ipdevpoll",
+        target_id="eventEngine",
+        event_type_id="juniperYellowAlarmState",
+        netbox=netbox,
+        state=Event.STATE_END,
+    )
+    event.varmap = {
+        "alerttype": "juniperYellowAlarmOff",
+    }
+    event.save()
+
+    return event
+
+
+def post_fake_red_alarm_start_event(netbox, count: int = 2):
+    event = Event(
+        source_id="ipdevpoll",
+        target_id="eventEngine",
+        event_type_id="juniperRedAlarmState",
+        netbox=netbox,
+        state=Event.STATE_START,
+    )
+    event.varmap = {
+        "count": count,
+        "alerttype": "juniperRedAlarmOn",
+    }
+    event.save()
+
+    return event
+
+
+def post_fake_red_alarm_end_event(netbox):
+    event = Event(
+        source_id="ipdevpoll",
+        target_id="eventEngine",
+        event_type_id="juniperRedAlarmState",
+        netbox=netbox,
+        state=Event.STATE_END,
+    )
+    event.varmap = {
+        "alerttype": "juniperRedAlarmOff",
+    }
+    event.save()
+
+    return event
+
+
+def post_fake_stateless_event(netbox):
+    event = Event(
+        source_id="ipdevpoll",
+        target_id="eventEngine",
+        event_type_id="juniperYellowAlarmState",
+        netbox=netbox,
+        state=Event.STATE_STATELESS,
+    )
+    event.save()
+
+    return event
+
+
+def get_eventengine_output(timeout=10):
+    """
+    Runs eventengine in foreground mode, kills it after timeout seconds and
+    returns the combined stdout+stderr output from the process.
+    Also asserts that pping shouldn't unexpectedly exit with a zero exitcode.
+    """
+    cmd = ["eventengine", "-f"]
+    try:
+        output = check_output(cmd, stderr=STDOUT, timeout=timeout)
+    except TimeoutExpired as error:
+        # this is the normal case, since we need to kill eventengine after the timeout
+        print(error.output.decode("utf-8"))
+        return error.output.decode("utf-8")
+    except CalledProcessError as error:
+        print(error.output.decode("utf-8"))
+        raise
+    else:
+        print(output)
+        assert False, "eventengine exited with non-zero status"
```

Comment:
This seems to be a duplicate of the fixture used in `boxdown_test.py`.  The fixture should be moved to `tests/integrationn/eventengine/conftest.py` so it can easily be re-used across any eventengine-related integration test.